src/parser/ast_common.c

```diff
@@ -8,7 +8,8 @@
 #include "./ast_common.h"
 #include "../value.h"
 
-AST_Variable* New_AST_Variable(const char* alias, const char* property) {
+// TODO I think a lot of unnecessary allocations here are unnecessary
```

Comment:
Why do you think we can use given label and alias without duplicating them?

Reply:
Sorry, that comment was unclear (also, really poor English, now that I'm looking at it!).

This function gets called for SET and ORDER clauses, as well as arithmetic expressions describing labels and/or aliases (which is used for RETURN clauses). A lot of those contexts then duplicate the label and alias again, which is the redundant step I would like to resolve.

Reply:
Redis manages all of the `argv` strings itself, and guarantees that they are not released until the calling client is unblocked. The server traceback looks like this:
```
==28665==    by 0x14F920: freeStringObject (object.c:244)
==28665==    by 0x14F920: decrRefCount (object.c:313)
==28665==    by 0x148E08: freeClientArgv (networking.c:726)
==28665==    by 0x149F3A: resetClient (networking.c:1045)
==28665==    by 0x19F76D: unblockClient (blocked.c:140)
==28665==    by 0x1AB55A: moduleHandleBlockedClients (module.c:3574)
==28665==    by 0x13A026: beforeSleep (server.c:1222)
==28665==    by 0x136B4D: aeMain (ae.c:500)
==28665==    by 0x133730: main (server.c:3899)
```
Since we'll never unblock a client until cleanup, I believe that pointers instead of dups for its component strings is safe!